Toolkit: docker image for plugin CI process (#22790)
grafana/grafana: packages/grafana-toolkit/docker/grafana-plugin-ci/README.md

> # Using this docker image

Reviewer: Maybe make it more clear that this docker image is intended for use in the plugin CI build systems.

Currently tagged and uploaded to dockerhub as srclosson/integrations-ci-build 

Based off of `circleci/node:12-browsers` 

## User
The user will be `circleci`
The home directory will be `/home/circleci`

## Node
- node 12 is installed
- yarn is installed globally
- npm is installed globally

## Go
- Go 1.14 is installed in `/usr/local/bin/go`

> - golangci 1.23.7 is installed in `/usr/local/bin/gofmt`

Reviewer 2: Should this be golangci-lint installed in `/usr/local/bin/golangci-lint`?

- mage is installed in `/home/circleci/go/bin/mage`

All of the above directories are in the path, so there is no need to specify fully qualified paths.

## Grafana source
- Installed in `/home/circleci/src/grafana`
- `yarn install` has been run

## Integration/Release Testing
There are 4 previous versions pre-downloaded to /usr/local/grafana. These versions are:
1. 6.6.2
2. 6.5.3
3. 6.4.5
4. 6.3.7

> To test, your circle-ci config will need a run section with something similar to the following

Reviewer 2: ```suggestion
To test, your CircleCI config will need a run section with something similar to the following:
```

```
- run:
        name: Setup Grafana (local install)
        command: |
          sudo dpkg -i /usr/local/grafana/deb/grafana_6.6.2_amd64.deb
          sudo cp ci/grafana-test-env/custom.ini /usr/share/grafana/conf/custom.ini
          sudo cp ci/grafana-test-env/custom.ini /etc/grafana/grafana.ini
          sudo service grafana-server start

>           sudo grafana-cli --version

Reviewer 2: Don't need superuser rights here?
```suggestion
          grafana-cli --version
```

```

# Building

> To build cd to `<srcroot>/packages/grafana-toolkit/docker/grafana-plugin-ci`

Reviewer 2: ```suggestion
To build, cd to `<srcroot>/packages/grafana-toolkit/docker/grafana-plugin-ci`
```

## Go
- Go 1.14 is installed in `/usr/local/bin/go`

> - golangci 1.23.7 is installed in `/usr/local/bin/golangci-lint`

Reviewer 2: ```suggestion
- golangci-lint 1.23.7 is installed in `/usr/local/bin/golangci-lint`
```

3. 6.4.5
4. 6.3.7

To test, your CircleCI config will need a run section with something similar to the following
```
- run:
        name: Setup Grafana (local install)
        command: |
          sudo dpkg -i /usr/local/grafana/deb/grafana_6.6.2_amd64.deb
          sudo updatedb

>           sudo locate grafana

Reviewer 2: Why are these commands necessary? It will just print the location of grafana?

Author: Well, they may not be, but at the time it was taking a long time to commit a new VM. Upon review, I had not updated the readme to match what I had tested. Yeah, this may not be necessary, but taking them both out caused an error, and in the interest of time decided not to pick that battle. I can still figure it out, I'm just competing for bandwidth with kids as they are all home youtubing :)
